lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -2037,7 +2037,8 @@ class ModuleWriter {
   }
 
   void writePrologue(raw_ostream &out) {
-    out << "// Generated by " << version::getSwiftFullVersion() << "\n"
+    out << "// Generated by " << version::getSwiftFullVersion(
+      M.getASTContext().LangOpts.EffectiveLanguageVersion) << "\n"
```

Comment:
I flip-flopped a few times, but now I agree: the printed headers from "Swift 3 mode" should be equivalent to what the Swift 3 compiler produced.